lib/clients/authenticated.js

```diff
@@ -1,332 +1,240 @@
     params.side = 'sell';
-    return self._placeOrder(params, callback);
-  };
+    return this._placeOrder(params, callback);
+  }
 
-  prototype.getTrailingVolume = function(callback) {
-    var self = this;
-    return prototype.get.call(self, ['users', 'self', 'trailing-volume'], {}, callback);
-  };
+  getTrailingVolume(callback) {
+    return this.get(['users', 'self', 'trailing-volume'], {}, callback);
+  }
 
-  prototype.cancelOrder = function(orderID, callback) {
-    var self = this;
+  cancelOrder(orderID, callback) {
 
-    if (!callback && (typeof orderID === 'function')) {
-      callback = orderID;
-      callback(new Error('must provide an orderID or consider cancelOrders'));
-      return;
+    if(!orderID || typeof orderID == 'function') {
+      let err = new Error('must provide an orderID or consider cancelOrders');
+      if(typeof orderID == 'function') orderID(err);
+      return Promise.reject(err);
     }
 
-    return prototype.delete.call(self, ['orders', orderID], callback);
-  };
+    return this.delete(['orders', orderID], callback);
+  }
 
-  prototype.cancelOrders = function(callback) {
-    var self = this;
-    return prototype.delete.call(self, ['orders'], callback);
-  };
+  cancelOrders(callback) {
+    return this.delete(['orders'], callback);
+  }
 
   // temp over ride public call to get Product Orderbook
-  prototype.getProductOrderBook = function(args, productId, callback) {
-    var self = this;
-
-    args = args || {}
-    if (!callback && (typeof args === 'function')) {
+  getProductOrderBook(args = {}, productId, callback) {
+    if (!callback && typeof args == 'function') {
       callback = args;
       args = {};
     }
 
-    var opts = { 'qs': args };
-    return prototype.get.call(self, ['products', productId, 'book'], opts, callback);
-  };
-
-
-  prototype.cancelAllOrders = function(args, callback) {
-    var self = this;
-    var currentDeletedOrders = [];
-    var totalDeletedOrders = [];
-    var query = true;
-    var response;
+    return this.get(['products', productId, 'book'], { qs: args }, callback);
+  }
 
-    args = args || {}
-    if (!callback && (typeof args === 'function')) {
+  async cancelAllOrders(args = {}, callback) {
+    if (!callback && typeof args == 'function') {
       callback = args;
       args = {};
     }
 
-    var opts = { 'qs': args };
+    if (!callback) callback = () => {};
 
-    async.doWhilst(
-      deleteOrders,
-      untilEmpty,
-      completed
-    );
+    const opts = { qs: args };
+    const totalDeletedOrders = [];
 
-    function deleteOrders(done) {
-      prototype.delete.call(self, ['orders'], opts, function(err, resp, data) {
+    let response, data;
 
-        if (err) {
-          done(err);
-          return;
-        }
+    try {
+      do {
+        // Have to use a special promise to get response as well as data
+        // for later passing to the callback;
+        [response, data] = await new Promise((resolve, reject) => {
```

Comment:
I'd prefer to not use async/await in this library for now (node@6 should be the target).